app/src/main/java/com/github/se/studybuddies/viewModels/UserViewModel.kt

```diff
@@ -12,53 +12,54 @@ import kotlinx.coroutines.launch
 import kotlinx.coroutines.tasks.await
 import kotlinx.coroutines.withContext
 
-class UserViewModel ( val uid: String? = null) : ViewModel() {
-    private val db = DatabaseConnection()
-    private val _userData = MutableLiveData<User>()
-    val userData: LiveData<User> = _userData
-    init {
-        if (uid != null) {
-            fetchUserData(getCurrentUserUID())
-            Log.d("MyPrint", "UserViewModel initialized with uid $uid")
-        } else {
-            Log.d("MyPrint", "UserViewModel initialized without uid")
-        }
-    }
+class UserViewModel(val uid: String? = null) : ViewModel() {
+  private val db = DatabaseConnection()
+  private val _userData = MutableLiveData<User>()
+  val userData: LiveData<User> = _userData
 
-    fun getCurrentUserUID(): String {
-        return db.getCurrentUserUID()
+  init {
+    if (uid != null) {
+      fetchUserData(getCurrentUserUID())
+      Log.d("MyPrint", "UserViewModel initialized with uid $uid")
+    } else {
+      Log.d("MyPrint", "UserViewModel initialized without uid")
     }
-    fun fetchUserData(uid: String) {
-        viewModelScope.launch {
-            try {
-                val document = db.getUserData(uid).await()
-                if (document.exists()) {
-                    val email = document.getString("email") ?: ""
-                    val username = document.getString("username") ?: ""
-                    val photoUrl = Uri.parse(document.getString("photoUrl") ?: "")
-                    val user = User(uid, email, username, photoUrl)
-                    _userData.value = user
-                } else {
-                    Log.d ("MyPrint", "In ViewModel, document not found")
-                    _userData.value = User.empty()
-                }
-            } catch (e: Exception) {
-                Log.d("MyPrint", "In ViewModel, failed to fetch user data with error: ", e)
-                _userData.value = User.empty()
-            }
-        }
-    }
-    suspend fun getDefaultProfilePicture(): Uri {
-        return withContext(Dispatchers.IO) {
-            db.getDefaultProfilePicture()
+  }
+
+  fun getCurrentUserUID(): String {
+    return db.getCurrentUserUID()
+  }
+
+  fun fetchUserData(uid: String) {
+    viewModelScope.launch {
+      try {
+        val document = db.getUserData(uid).await()
+        if (document.exists()) {
```

Comment:
Same as in `GroupViewModel`, I think extraction should be performed directly in the `DatabaseConnection` class. What do you think ?